### PR TITLE
Fix Content-Type custom payload bugs

### PR DESCRIPTION
**Pull request by marina-p**
When a custom payload body is used to work around unsupported content types, the Content-Type header must also be modified via the dictionary.  This functionality had several bugs, which are fixed in this change.  After this update, using either `restler_custom_payload_header` or `restler_custom_payload` with the request-specific syntax for dictionary payloads now works as expected, for example:

  "restler_custom_payload": {
    "/stores/{storeId}/order/post/Content-Type": [ "xml" ]
  }

This change also fixes the same bugs with per-request syntax with restler_custom_payload_query.

Testing:
- manual testing with demo_server
- added new compiler tests

### CODE REVIEW

File: src/compiler/Restler.Compiler/Dictionary.fs [hunk 1]
```diff
@@ -58,8 +58,18 @@ type MutationsDictionary =
...
             | None -> None
 
+        member x.getRequestTypePayloadPrefix endpoint (method:string) = 
+            sprintf "%s/%s/" endpoint (method.ToLower())
+
+        member x.getRequestTypePayloadName endpoint (method:string) propertyNameOrPath = 
+            sprintf "%s/%s/%s" endpoint (method.ToLower()) propertyNameOrPath
+
+        member x.isRequestTypePayloadName endpoint (method:string) (propertyValue:string) = 
+             let prefix = sprintf "%s/%s" endpoint (method.ToLower())
```
Comment (wilbaker) [inline src/compiler/Restler.Compiler/Dictionary.fs:68]: Should `prefix` include a trailing `/` like it does here in `getRequestTypePayloadPrefix`?

```
        member x.getRequestTypePayloadPrefix endpoint (method:string) = 
            sprintf "%s/%s/" endpoint (method.ToLower())
```

Could there be an issue here with `propertyValue.StartsWith(prefix)` incorrectly matching with a `prefix` that's actually a substring of `propertyValue`?

prefix = "foo/bar"
propertyValue = "foo/bar2"

File: src/compiler/Restler.Compiler/Compiler.fs [hunk 2]
```diff
@@ -858,15 +858,59 @@ let generateRequestPrimitives (requestId:RequestId)
...
-            // Filter out Content-Type, because this is handled separately below
-            let injectedCustomPayloadHeaderParameters = getInjectedCustomPayloadParameters dictionary CustomPayloadType.Header ["Content-Type"]
+            // No headers in Swagger
+            let injectedCustomPayloadHeaderParameters = getInjectedHeaderOrQueryParameters [] CustomPayloadType.Header
             [(ParameterPayloadSource.DictionaryCustomPayload, ParameterList (injectedCustomPayloadHeaderParameters |> seq))]
+        | [(Schema, ParameterList y)] when y |> Seq.isEmpty ->
+            // No headers in Swagger
+            let injectedCustomPayloadHeaderParameters = getInjectedHeaderOrQueryParameters [] CustomPayloadType.Header
+            if injectedCustomPayloadHeaderParameters |> Seq.isEmpty then
+                // TODO: remove this branch and update grammar.py test baselines
```
Comment (wilbaker) [inline src/compiler/Restler.Compiler/Compiler.fs:910]: Just wanted to check if this TODO is for this PR, or if it's to be addressed in future set of changes.

**Review by wilbaker — CHANGES_REQUESTED** [review]
Wanted to make sure the prefix check in Dictionary.fs is correct

File: src/compiler/Restler.Compiler/Compiler.fs (hunk 2 above)
Comment (marina-p) [reply src/compiler/Restler.Compiler/Compiler.fs:910]: It's for a future set of changes, updating the grammar.json files is a bit time consuming - in addition to the compiler updates, need to make sure the engine can consume the new grammar.json files and this particular change could introduce a regression.

**Review by wilbaker — APPROVED** [review]
Changes look good to me. FYI that I have not tested/validated these changes myself. Thanks!

**Pull request merged**